### ISSUE TITLE
writeWithResponse bug?

### ISSUE THREAD

**Issue by sujal** [OpenBLE/OpenBLE#6]
Hi there. Looking at line 243 (in  9b7cd25 -> https://github.com/OpenBLE/OpenBLE/blob/master/Classes/LeDataService.m#L243), I see the following:

```
if([writeWithResponseCBUUIDs containsObject:writeCharacteristic])
```

Shouldn't that be checking `writeCharacteristic.UUID`?

I tested this change locally and it seemed to fix my write issues with my BLE Shield (a RedBear v.2012.07). Making that change correctly identified the characteristic as a write without response call.

Also, as a side note, shouldn't `writeWithResponseCBUUIDs` and related variables be called `writeWithoutResponseCBUUIDs`? The fact that it's called writeWITH totally confused me while debugging my issue...

**Comment by jacobrosenthal** [comment]
I meant to test on redbear before I 1.0.0 but that didnt happen. Ill take a
look at this in another day or so when Im off another project.

Thanks for the bug report!

**Comment by jacobrosenthal** [comment]
Yeah I recall patching something in there last minute and holy shit its terrible right now.
Can you take a look at this fix

**Comment by digiponta** [comment]
I can't use the openBLE for BLE mini, maybe for the same issue with iOS 7.1 and BLE mini. I had escaped the issue by modifying the BLE.m to "writewithresponce", then iOS seems to execute a function with only an error but  a warning.

**Comment by digiponta** [comment]
Thank you that I had known the bug, I said, had been fixed, by checking a new one that I had bought currently. I would like to know where a new firmware I need for the old one.

**Comment by bairesearch** [comment]
Regarding https://github.com/jacobrosenthal/OpenBLE/commit/3f1b77f547c4adfcce2ccd7a29b9b2a284f52a68:

Note as per the pull suggestion comments (https://github.com/OpenBLE/OpenBLE/pull/7), there is a typo in the commit (Classes/LeDataService.m) that causes write without response to fail;

Change (second instance [line 83]);
writeWithResponseCBUUIDs=[[NSMutableArray alloc] init]; 
to;
writeWithoutResponseCBUUIDs=[[NSMutableArray alloc] init];

(see https://github.com/baxterai/OpenBLE/commit/6c5ccfe56c15530ce4221efefef7715e871f5750).

---
There is also a limitation in the write implementation which has not been fixed by this commit (and should be patched later);

Write does not work if the write and read characteristics UUIDs are identical (see https://github.com/baxterai/OpenBLE/commit/3ebc891fd89b395c82326ec99197ba9f8fa9f57c for fix).
